tests/png/bug00295.c

```diff
@@ -0,0 +1,39 @@
+/**
+ * Regression test for <https://github.com/libgd/libgd/issues/295>.
+ *
+ * We're testing that reading a PNG which might trigger a libpng warning
+ * does not trigger a GD_ERROR.
+ */
+
+
+#include "gd.h"
+#include "gd_errors.h"
+#include "gdtest.h"
+#include <string.h>
+
+
+static void error_handler(int priority, const char *format, va_list args)
+{
+    char msg[256];
+
+    vsprintf(msg, format, args);
+    gdTestAssertMsg(priority != GD_ERROR,
+                    "unexpected GD_ERROR: %s\n", msg);
+}
+
+
+int main()
+{
+    gdImagePtr im;
+    FILE *fp;
+
+    gdSetErrorMethod(error_handler);
+    fp = gdTestFileOpen2("png", "bug00295.png");
+
+    im = gdImageCreateFromPng(fp);
+
+    fclose(fp);
+    gdImageDestroy(im);
+
+    return gdNumFailures();

```

Comment:
can we do a version check ?  if it's less than 1.6, don't require it to be set, but if it's >=1.6, require it ?

can we feed it a more damaged image that'll more strongly guarantee libpng calls us ?